src/Core/src/Handlers/Button/ButtonHandler.Android.cs

```diff
@@ -185,7 +187,9 @@ public override void SetImageSource(Drawable? platformImage)
 				if (Handler?.PlatformView is not MaterialButton button)
 					return;
 
-				button.Icon = platformImage;
+				button.Icon = platformImage is null
+					? null
+					: new MauiMaterialButton.MauiResizableDrawable(platformImage);
```

Comment:
When we set the icon, we wrap it in a special "resizable drawable" that allows us to give the icon a specific size that is used by the measure and layout passes.